assert/assertions.go

```diff
@@ -1194,6 +1194,10 @@ func InDeltaSlice(t TestingT, expected, actual interface{}, delta float64, msgAn
 	actualSlice := reflect.ValueOf(actual)
 	expectedSlice := reflect.ValueOf(expected)
 
+	if expectedSlice.Len() < actualSlice.Len() {
```

Comment:
Taking line 1225 below as an example, `InDeltaMapValues` tests that the expected and actual maps are the same length, and then because of line 1237 that the maps have identical keys.

I think it would make assert more consistent if this tested that both slices have equal length. The downside is that this could cause tests that currently pass to fail.